textx/model.py

```diff
@@ -718,16 +773,21 @@ def call_obj_processors(metamodel, model_obj,
             if current_metaclass_of_obj._tx_fqn !=\
                     metaclass_of_grammar_rule._tx_fqn:
                 assert RULE_ABSTRACT == metaclass_of_grammar_rule._tx_type
-                obj_processor_current = metamodel.obj_processors.get(
-                    current_metaclass_of_obj.__name__, None)
-                if obj_processor_current:
-                    return_value_current = obj_processor_current(model_obj)
+                return_value_current = metamodel.process(
+                    model_obj, current_metaclass_of_obj.__name__,
+                    **get_location(model_obj))
 
         # call obj_proc of rule found in grammar
-        obj_processor_grammar = metamodel.obj_processors.get(
-            metaclass_of_grammar_rule.__name__, None)
-        if obj_processor_grammar:
-            return_value_grammar = obj_processor_grammar(model_obj)
+        if metamodel.has_obj_processor_for_type(metaclass_of_grammar_rule.__name__):
+            try:
+                loc = get_location(model_obj)
+            except Exception:
+                raise TextXError("you may not define an object processor for "
+                                 "a class '{}' preventing to get the location "
+                                 "from.".format(metaclass_of_grammar_rule.__name__))
```

Comment:
This error message is not clear to me.

Reply:
Initially I had not the "if obj_processor is defined" statement above - and always called the object_processor for the current element. It then happened, that the object processor was called for an "int" or "str" causing an error.

I do not think we run into that case with the additional if. So I removed the try-except-block (and the confusing error message).

Note: if the error happens, we get an exception anyhow... So no need to transform this error here!